### PR TITLE
fix escaping of llvm-cov output

### PR DESCRIPTION
**Pull request by davidhewitt**
Looks like the output of `cargo llvm-cov show-env` changed from escaped by `"` to `'`. I think that's the cause of our CI issues.

### CODE REVIEW

**Comment by davidhewitt** [comment]
I'll also add updates for Rust 1.72 to this PR...

File: noxfile.py [hunk 1]
```diff
@@ -606,7 +606,13 @@ def _get_coverage_env() -> Dict[str, str]:
 
     for line in output.strip().splitlines():
         (key, value) = line.split("=", maxsplit=1)
-        env[key] = value.strip('"')
+        # Strip single or double quotes from the variable value
+        # - quote used by llvm-cov differs between Windows and Linux
+        if value and value[0] == "'":
```
Comment (adamreichold) [inline noxfile.py:611]: This looks overly complex because the two branches are the same?

Comment (adamreichold) [reply noxfile.py:611]: Meaning

```python
if value and value[0] in ["'", '"']:
  value = value[1:-1]
```

should also work?

**Review by adamreichold — APPROVED** [review]
LGTM modulo `clippy::if_same_then_else` on the Python code.

File: noxfile.py (hunk 1 above)
Comment (davidhewitt) [reply noxfile.py]: Good point, fixed in force-push.

**Pull request merged**